packages/ember-principled-forms/tests/integration/components/form-field-test.ts

```diff
@@ -47,5 +61,47 @@ module('Integration | Component | form-field', function(hooks) {
     `);
 
     assert.equal(this.element.textContent!.trim(), label, 'yields field model back out');
+
+    await a11yAudit();
+  });
+
+  test('error state is correct', async function(assert) {
+    const name = Field.validate(Field.required<string>());
+
+    const noop = () => {};
+    this.setProperties({ name, onInput: noop, onChange: noop });
+
+    await render(hbs`{{form-field label='Name' model=this.name onInput=this.onInput}}`);
+    const nameInput = (this.element.querySelector('input') as HTMLInputElement);
+    const errorSpan = (this.element.querySelector('span') as HTMLElement);
+
+    assert.ok(errorSpan, 'error exists');
+    assert.ok(nameInput.getAttribute('aria-invalid'), 'invalid input has aria-invalid true');
+    assert.equal(nameInput.getAttribute('aria-describedby'), errorSpan.id, 'valid input has error id described-by');
+    assert.equal(errorSpan.textContent!.trim(), 'field is required', 'displays required error');
+
+    await a11yAudit();
+
+    // Block Format
+    await render(hbs`
+      {{#form-field label='Name' model=this.name onInput=this.onInput as |field|}}
+        <input
+          id={{field.id}}
+          value={{field.model.value}}
+          aria-invalid={{field.ariaInvalid}}
+          aria-describedby={{field.ariaDescribedBy}}
+        />
+
+        {{#if field.isInvalid}}
+          <span class={{errorClass}} id={{field.errorId}}>{{field.model.validity.reason}}</span>
+        {{/if}}
+      {{/form-field}}
+    `);
+
+    const blockNameInput = (this.element.querySelector('input') as HTMLInputElement);
+    const blockErrorSpan = (this.element.querySelector('span') as HTMLElement);
+    assert.ok(blockNameInput.getAttribute('aria-invalid'), 'block | invalid input has aria-invalid true');
+    assert.equal(blockNameInput.getAttribute('aria-describedby'), blockErrorSpan.id, 'block | valid input has error id described-by');
+    assert.equal(blockErrorSpan.textContent!.trim(), 'field is required', 'block | displays required error');
```

Comment:
Didn't bother to `a11yAudit` here since just testing the properties can be used in block form the actual template is not accessible.